src/LiveComponent/src/Util/UrlFactory.php

```diff
@@ -60,10 +54,24 @@ public function createFromPreviousAndProps(
 
     private function createPath(string $previousUrl, array $props): string
     {
-        return $this->router->generate(
-            $this->router->match($previousUrl)['_route'] ?? '',
+        $newPath = $this->router->generate(
+            $this->matchRoute($previousUrl),
             $props
         );
+
+        return $newPath;
+    }
+
+    private function matchRoute(string $previousUrl): string
+    {
+        $context = $this->router->getContext();
+        $tmpContext = clone $context;
+        $tmpContext->setMethod('GET');
+        $this->router->setContext($tmpContext);
+        $match = $this->router->match($previousUrl);
+        $this->router->setContext($context);
```

Comment:
So the previous context can be re-set correctly:
```suggestion
        $context = $this->router->getContext();
        $tmpContext = clone $context;
        $tmpContext->setMethod('GET');
        $this->router->setContext($tmpContext);
        try {
            $match = $this->router->match($previousUrl);
        } finally {
            $this->router->setContext($context);
        }
```